gradle-baseline-java/src/main/groovy/com/palantir/baseline/extensions/BaselineJavaVersionExtension.java

```diff
@@ -26,19 +31,33 @@
  * target and runtime java versions used for a single project.
  */
 public class BaselineJavaVersionExtension {
+    private static final Pattern ZULU_VERSION_PATTERN = Pattern.compile("ZULU_([\\d]+)_VERSION");
+    private static final Pattern JAVA_VERSION_PATTERN = Pattern.compile("JAVA_([\\d]+)_VERSION");
 
     private final Property<JavaLanguageVersion> target;
     private final Property<JavaLanguageVersion> runtime;
     private final Property<Boolean> overrideLibraryAutoDetection;
+    private final MapProperty<JavaLanguageVersion, String> zuluVersions;
+    private final MapProperty<JavaLanguageVersion, String> javaVersions;
 
     @Inject
     public BaselineJavaVersionExtension(Project project) {
         target = project.getObjects().property(JavaLanguageVersion.class);
-        runtime = project.getObjects().property(JavaLanguageVersion.class);
-        overrideLibraryAutoDetection = project.getObjects().property(Boolean.class);
         target.finalizeValueOnRead();
+
+        runtime = project.getObjects().property(JavaLanguageVersion.class);
         runtime.finalizeValueOnRead();
+
+        overrideLibraryAutoDetection = project.getObjects().property(Boolean.class);
         overrideLibraryAutoDetection.finalizeValueOnRead();
+
+        zuluVersions = project.getObjects().mapProperty(JavaLanguageVersion.class, String.class);
+        zuluVersions.putAll(project.provider(() -> parseOutVersions(project, ZULU_VERSION_PATTERN)));
+        zuluVersions.finalizeValueOnRead();
+
+        javaVersions = project.getObjects().mapProperty(JavaLanguageVersion.class, String.class);
+        javaVersions.putAll(project.provider(() -> parseOutVersions(project, JAVA_VERSION_PATTERN)));
+        javaVersions.finalizeValueOnRead();
```

Comment:
Do you think we could define these at most once per repo instead of per-module? (possibly `BaselineJavaVersionsExtension.java`, which only exists on the root project)
I don't think we would want the values to differ across modules.

Reply:
Heh actually in the process of doing this right now. The `BaselineJavaVersion` vs `BaselineJavaVersions` difference was lost on me for a little bit.